compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -693,7 +694,7 @@ abstract class PrepJSInterop extends plugins.PluginComponent
         }
 
         val ownerLoadSpec = jsInterop.jsNativeLoadSpecOf(sym.owner)
-        val jsName = jsInterop.jsNameOf(sym)
+        val JSName.Literal(jsName) = jsInterop.jsNameOf(sym)
```

Comment:
In fact I realized that that it *needs* to be in the IR. Otherwise we cannot implement `java.lang.Class.isInstance` for these classes. Since there's no language reason not to support it, we need to.

So I'll revert to the IR changes you had made.